src/main/java/org/broadinstitute/hellbender/tools/coveragemodel/germline/CoverageModellerGermlineSparkToggle.java

```diff
@@ -0,0 +1,363 @@
+            ex.printStackTrace();
+            throw new UserException.CouldNotReadInputFile("Could not parse the germline contig ploidy annotations table");
+        }
+
+        logger.info("Parsing the copy number transition prior table and initializing the caches...");
+        final IntegerCopyNumberTransitionProbabilityCacheCollection transitionProbabilityCacheCollection;
+        try (final Reader copyNumberTransitionPriorTableReader = getReaderFromURI(copyNumberTransitionPriorTableURI)) {
+            final String parentURI = new File(copyNumberTransitionPriorTableURI).getParent();
+            transitionProbabilityCacheCollection = new IntegerCopyNumberTransitionProbabilityCacheCollection(
+                    copyNumberTransitionPriorTableReader, parentURI, true);
+        } catch (final IOException ex) {
+            ex.printStackTrace();
+            throw new UserException.CouldNotReadInputFile("Could not parse the copy number transition prior table");
+        }
+        final IntegerCopyNumberExpectationsCalculator integerCopyNumberExpectationsCalculator =
+                new IntegerCopyNumberExpectationsCalculator(transitionProbabilityCacheCollection,
+                        params.getMinLearningReadCount());
+
+        final CoverageModelParameters model;
+        if (modelPath != null) {
+            logger.info("Loading model parameters...");
+            model = CoverageModelParameters.read(modelPath);
+        } else {
+            model = null;
+        }
+
+        logger.info("Initializing the EM algorithm workspace...");
+        final CoverageModelEMWorkspaceGermline workspace = new CoverageModelEMWorkspaceGermline(
+                readCounts, ploidyAnnotatedTargetCollection, sexGenotypeDataCollection,
+                integerCopyNumberExpectationsCalculator, params, model, ctx);
+        final CoverageModelEMAlgorithm<IntegerCopyNumberState> algo = new CoverageModelEMAlgorithm<>(params,
+                workspace);
+
+        switch (jobType) {
+            case LEARN_AND_CALL:
+                algo.runExpectationMaximization();
+                logger.info("Saving the model to disk...");
+                workspace.saveModel(new File(outputPath, FINAL_MODEL_SUBDIR).getAbsolutePath());
+                break;
+
+            case CALL_ONLY:
+                algo.runExpectation();
+        }
+
+        logger.info("Saving posteriors to disk...");
+        workspace.savePosteriors(new File(outputPath, FINAL_POSTERIORS_SUBDIR).getAbsolutePath(),
+                PosteriorVerbosityLevel.EXTENDED);
+    }
+
+    /**
+     * Takes a URI string (which could a local file, a Google bucket file, or a file living on HDFS) and
+     * returns a {@link Reader}
+     *
+     * @param inputURI input URI string
+     * @return an instance of {@link Reader}
+     * @throws IOException if the {@link Reader} could not be instantiated
+     */
+    private Reader getReaderFromURI(@Nonnull final String inputURI) throws IOException {
```

Comment:
Actually, `BucketUtils.openFile(String, PipelineOptions)` does all of `getReaderFromURI(String)` except for it returns an `InputStream` and requires `PipelineOptions`. The latter is provided by `SparkCommandLineProgram`.

I simplified the method to:
```
    private Reader getReaderFromURI(@Nonnull final String path) {
        final InputStream inputStream = BucketUtils.openFile(path, getAuthenticatedGCSOptions());
        return new BufferedReader(new InputStreamReader(inputStream));
    }
```

I think the right habitat for this method is in gatk public's `SparkCommandLineProgram` but for now, I'm comfortable with having this little private method in `CoverageModellerGermlineSparkToggle` unless you have a better suggestion.